servers/navigation_server_2d.h

```diff
@@ -262,10 +276,15 @@ class NavigationServer2D : public Object {
 	virtual void obstacle_set_paused(RID p_obstacle, bool p_paused) = 0;
 	virtual bool obstacle_get_paused(RID p_obstacle) const = 0;
 	virtual void obstacle_set_radius(RID p_obstacle, real_t p_radius) = 0;
+	virtual real_t obstacle_get_radius(RID p_obstacle) const = 0;
 	virtual void obstacle_set_velocity(RID p_obstacle, Vector2 p_velocity) = 0;
+	virtual Vector2 obstacle_get_velocity(RID p_obstacle) const = 0;
 	virtual void obstacle_set_position(RID p_obstacle, Vector2 p_position) = 0;
+	virtual Vector2 obstacle_get_position(RID p_obstacle) const = 0;
 	virtual void obstacle_set_vertices(RID p_obstacle, const Vector<Vector2> &p_vertices) = 0;
+	virtual Vector<Vector2> obstacle_get_vertices(RID p_obstacle) const = 0;
```

Comment:
This signature is not a const ref because it is difficult to implement (since internally the 2D server is a wrapper over the 3D server, to return this by ref would have to dynamically allocate the result of a conversion from `Vector<Vector3>` to `Vector<Vector2>`, otherwise it would be a dangling ref.

I'm not sure how much the const ref would help/how it translates for the engine though.

Reply:
That is ok, I don't think we should use ref in that case anyway, the servers normally do not return refs directly to users.